homeassistant/components/totalconnect/button.py

```diff
@@ -0,0 +1,106 @@
+"""Interfaces with TotalConnect buttons."""
+
+from homeassistant.components.button import ButtonEntity, ButtonEntityDescription
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import EntityCategory
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.device_registry import DeviceInfo
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+
+from .const import DOMAIN
+
+
+async def async_setup_entry(
+    hass: HomeAssistant, entry: ConfigEntry, async_add_entities: AddEntitiesCallback
+) -> None:
+    """Set up TotalConnect buttons based on a config entry."""
+    buttons: list = []
+
+    client_locations = hass.data[DOMAIN][entry.entry_id].client.locations
+
+    for location_id, location in client_locations.items():
+        buttons.append(TotalConnectClearBypassButton(location))
+        buttons.append(TotalConnectBypassAllButton(location))
+
+        buttons.extend(
+            TotalConnectZoneBypassButton(location_id, zone)
+            for zone in location.zones.values()
+            if zone.can_be_bypassed
+        )
+
+    async_add_entities(buttons)
+
+
+class TotalConnectZoneBypassButton(ButtonEntity):
+    """Represent an TotalConnect zone bypass button."""
+
+    def __init__(self, location_id, zone):
+        """Initialize the TotalConnect status."""
+        self._zone = zone
+        self.entity_description: ButtonEntityDescription = ButtonEntityDescription(
+            key="bypass", name="Bypass", entity_category=EntityCategory.DIAGNOSTIC
+        )
+        self._attr_name = f"{zone.description} {self.entity_description.name}"
+        self._attr_unique_id = (
+            f"{location_id}_{zone.zoneid}_{self.entity_description.key}"
+        )
+
+    def press(self):
+        """Press the bypass button."""
+        self._zone.bypass()
+
+    @property
+    def device_info(self) -> DeviceInfo:
+        """Return device info."""
+        identifier = self._zone.sensor_serial_number or f"zone_{self._zone.zoneid}"
+        return DeviceInfo(
+            name=self._zone.description,
+            identifiers={(DOMAIN, identifier)},
+            serial_number=self._zone.sensor_serial_number,
+        )
```

Comment:
```suggestion
    """Represent a TotalConnect zone bypass button."""

    _attr_has_entity_name = True
    _attr_translation_key = "bypass"
    _attr_entity_category = EntityCategory.DIAGNOSTIC

    def __init__(self, location_id, zone) -> None:
        """Initialize the TotalConnect status."""
        self._zone = zone
        identifier = self._zone.sensor_serial_number or f"zone_{self._zone.zoneid}"
        self._attr_device_info = DeviceInfo(
            identifiers={(DOMAIN, identifier)},
            name=self._zone.description,
            serial_number=self._zone.sensor_serial_number,
        )
        self._attr_unique_id = f"{location_id}_{zone.zoneid}_bypass"

    def press(self):
        """Press the bypass button."""
        self._zone.bypass()
```

Reply:
When can the serial number be None?

Reply:
Some of the other older TotalConnect hardware does not report a serial number.  The `zoneid` is unique (but just a database key).